x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/EvalOperator.java

```diff
@@ -48,6 +47,11 @@ public String toString() {
     }
 
     public interface ExpressionEvaluator {
+
+        interface ExpressionEvaluatorFactory {
```

Comment:
I'd prefer to either make this a top level of call it `ExpressionEvaluator.Factory`. But, whatever.

Reply:
I called it `ExpressionEvaluator.Factory` (as you suggested) - since the factory is a nested type in similar areas too.